org.knime.knip.core/src/org/knime/knip/core/ui/imgviewer/panels/BrightnessContrastPanel.java

```diff
@@ -0,0 +1,852 @@
+ *  may freely choose the license terms applicable to such Node, including
+ *  when such Node is propagated with or for interoperation with KNIME.
+ * ---------------------------------------------------------------------
+ *
+ * Created on 01.12.2015 by oole
+ */
+package org.knime.knip.core.ui.imgviewer.panels;
+
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+import javax.swing.Box;
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JSlider;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+
+import org.knime.knip.core.KNIPGateway;
+import org.knime.knip.core.ui.event.EventListener;
+import org.knime.knip.core.ui.event.EventService;
+import org.knime.knip.core.ui.imgviewer.ViewerComponent;
+import org.knime.knip.core.ui.imgviewer.events.BrightnessContrastChgEvent;
+import org.knime.knip.core.ui.imgviewer.events.ImgAndLabelingChgEvent;
+import org.knime.knip.core.ui.imgviewer.events.ImgRedrawEvent;
+import org.knime.knip.core.ui.imgviewer.events.ImgWithMetadataChgEvent;
+import org.knime.knip.core.ui.imgviewer.events.PlaneSelectionEvent;
+import org.knime.knip.core.ui.imgviewer.panels.providers.AWTImageProvider;
+
+import net.imagej.ops.OpService;
+import net.imagej.widget.HistogramBundle;
+import net.imglib2.FinalInterval;
+import net.imglib2.IterableInterval;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.histogram.BinMapper1d;
+import net.imglib2.histogram.Histogram1d;
+import net.imglib2.histogram.Real1dBinMapper;
+import net.imglib2.img.Img;
+import net.imglib2.ops.operation.real.unary.Normalize;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.view.Views;
+
+/**
+ * Adjust Brightness and Contrast of an image.
+ *
+ * @author <a href="mailto:ole.c.ostergaard@gmail.com">Ole Ostergaard</a>
+ * @param <T>
+ * @param <I>
+ */
+public class BrightnessContrastPanel<T extends RealType<T>, I extends Img<T>> extends ViewerComponent {
+
+    /* slider values */
+    private static final int SLIDER_MIN = 0;
+    private static final int SLIDER_MAX = 100;
+    private static final long serialVersionUID = 1L;
+
+    /* sliders */
+    private JSlider m_minimumSlider;
+    private JSlider m_maximumSlider;
+    private JSlider m_brightnessSlider;
+    private JSlider m_contrastSlider;
+
+    /* buttons */
+    private JButton m_automaticSaturationButton;
+    private JButton m_resetButton;
+
+    /* checkbox */
+    private JCheckBox m_autoSelect;
+    private JCheckBox m_planeSelect;
+
+    /* labels */
+    private JLabel m_min;
+    private JLabel m_max;
+    private JLabel m_bright;
+    private JLabel m_contrast;
+    private JLabel w_minLabel;
```

Comment:
Please be consistent with the naming of variables:
- always use the ``m_`` prefix for members.
- avoid other prefixes like the ``w_`` one used here.